CONTRIBUTING.md

```diff
@@ -0,0 +1,65 @@
+# Contributing to Yearn
+This is a living guide that defines a development process for The Yearn.finance Core Protocol. 
+
+We want to make contributing to this project as easy and transparent as possible, whether it's:
+
+- Reporting a bug
+- Discussing the current state of the code
+- Submitting a fix
+- Proposing new features
+- Becoming a maintainer
+
+## We Develop with Github
+We use github to host code, to track issues and feature requests, as well as accept pull requests.
+
+## We Use [Github Flow](https://guides.github.com/introduction/flow/index.html), So All Code Changes Happen Through Pull Requests
+Pull requests are the best way to propose changes to the codebase (we use [Github Flow](https://guides.github.com/introduction/flow/index.html)). We actively welcome your pull requests:
+
+1. Fork the repo and create your branch from `master`.
+2. If you've added code that should be tested, add tests or ensure it doesn't break current tests.
+3. If you've changed something impacting current docs, please update the documentation.
+4. Ensure the test suite passes if applicable.
+5. Make sure your code lints.
+6. Issue that pull request!
+
+## Release Process
+The `master` branch has the up to date changes of the codebase with working code. Releases should be tracked via git tags to link a specific commit to a deployment for history and documentation purposes.
+
+## Github Actions
+Repository uses GH actions to setup CI for test harness.
+You can see an example [here](https://github.com/iearn-finance/yearn-protocol/blob/master/.github/workflows/test.yaml)
+
+Be sure to setup in the repository any secret or env variable needed for your GH actions to work.
+
+## Any contributions you make will be under the APGL Software License
+In short, when you submit code changes, your submissions are understood to be under the same [APGL License](https://choosealicense.com/licenses/agpl-3.0/) that covers the project. Feel free to contact the maintainers if that's a concern.
+
+## Report bugs using Github's [issues](https://github.com/iearn-finance/yearn-protocol/issues)
+We use GitHub issues to track public bugs. Report a bug by [opening a new issue](https://github.com/iearn-finance/yearn-protocol/issues/new); it's that easy!
+
+Before adding a new issue, please check that your issue is not already identified or hasn't been handled by searching the active/closed issues.
+
+## Write bug reports with detail, background, and sample code
```

Comment:
You can add templates for bug reports and pull requests.

Here are some examples:
https://github.com/vyperlang/vyper/tree/master/.github/ISSUE_TEMPLATE
https://github.com/vyperlang/vyper/blob/master/.github/PULL_REQUEST_TEMPLATE.md